### PR TITLE
feat: SDK docs

### PR DESCRIPTION
**Pull request by cjbell**
### Description

The goal of this PR is to introduce a new set of sections to the docs to cover our SDKs. The purpose of doing so is to better separate out SDK documentation and quick-start guides from where they live currently, which is under the "Building in-app UI" section. This PR introduces:

- A new SDK overview section that links to all available SDKs
- New SDK pages for Javascript, React ,React Native, iOS, Android, Flutter
- Reworks the building in-app UI pages for Javascript, React Native, iOS, Android, Flutter
- New `Card` and `SdkCard` components

I also took the opportunity in this PR to clean up some of our old layout code and reorganize things to be cleaner, with fewer layout wrappers, making things easier to work with going forward.

### Screenshots

image

### CODE REVIEW

**Comment by matgar26** [comment]
@cjbell I think this all looks great!

**Review by matgar26 — APPROVED** [review]

File: content/in-app-ui/ios/overview.mdx [hunk 1]
```diff
@@ -1,33 +1,60 @@
...
+  text={
+    <>
+      <span className="font-bold">Note</span> in the current version of the
+      Swift SDK there are no pre-built UI elements.
+    </>
+  }
+/>
 
-**Quick links**
+The Knock iOS SDK is a low-level set of methods for interacting with the Knock APIs from iOS applications. The SDK is designed to help you easily integrate Knock into your application and build in-app notification experiences powered by Knock.
```
Comment (JEverhart383) [inline content/in-app-ui/ios/overview.mdx:17]:
```suggestion
The Knock Swift SDK is a low-level set of methods for interacting with the Knock APIs from iOS and macOS applications. The SDK is designed to help you easily integrate Knock into your application and build in-app notification experiences powered by Knock.
```
Just echoing language from the page meta and what the GH repo calls itself

File: content/in-app-ui/javascript/overview.mdx [hunk 2]
```diff
@@ -1,34 +1,37 @@
 ---
-title: Building in-app UI in Javascript
-description: Learn more about the in-app notifications experiences you can build in Javascript with Knock. Easily power notification feed, toasts, and inbox experiences with our client-side SDKs.
+title: "Building in-app UI in Javascript (Web)"
```
Comment (JEverhart383) [inline content/in-app-ui/javascript/overview.mdx:2]:
```suggestion
title: "Building in-app UI in JavaScript (Web)"
```

File: content/in-app-ui/javascript/overview.mdx [hunk 3]
```diff
@@ -1,34 +1,37 @@
...
+    <>
+      If you're looking to use pre-built in-app UI elements for a web
+      application, you can look at{" "}
+      <a href="/in-app-ui/react">React components</a>.
+    </>
+  }
+/>
 
-**Note**: if you're already using React in your application, you might want to take a look at our [React library for building in-app UI](/in-app-ui/react/overview).
+The Knock Javascript client SDK is a low-level set of methods for interacting with the Knock APIs from client-side Javascript web applications. The SDK is designed to help you easily integrate Knock into your application and build in-app notification experiences powered by Knock.
```
Comment (JEverhart383) [inline content/in-app-ui/javascript/overview.mdx:18]:
```suggestion
The Knock JavaScript client SDK is a low-level set of methods for interacting with the Knock APIs from client-side JavaScript web applications. The SDK is designed to help you easily integrate Knock into your application and build in-app notification experiences powered by Knock.
```

File: content/in-app-ui/react/overview.mdx
Comment (JEverhart383) [inline content/in-app-ui/react/overview.mdx:1]: I believe we should keep this callout regarding the deprecation of the old package. We still get people in Slack or elsewhere who haven't upgraded

File: content/sdks/javascript/reference.mdx [hunk 4]
```diff
@@ -1,8 +1,8 @@
 ---
-title: "API Reference (@knocklabs/client)"
-description: Complete API reference for the Knock JavaScript Client SDK.
+title: "Javascript SDK API Reference"
```
Comment (JEverhart383) [inline content/sdks/javascript/reference.mdx:2]:
```suggestion
title: "JavaScript SDK API Reference"
```

File: content/sdks/javascript/reference.mdx [hunk 5]
```diff
@@ -1,8 +1,8 @@
...
+description: Complete API reference for the Knock Javascript SDK.
```
Comment (JEverhart383) [inline content/sdks/javascript/reference.mdx:3]:
```suggestion
description: Complete API reference for the Knock JavaScript SDK.
```

File: content/sdks/overview.mdx
Comment (JEverhart383) [inline content/sdks/overview.mdx:1]: Overall, love the visual feel of this page

File: content/sdks/overview.mdx [hunk 6]
```diff
@@ -0,0 +1,92 @@
...
+  <SdkCard
+    title="PHP SDK"
+    linkUrl="https://github.com/knocklabs/knock-php"
+    languages={["PHP"]}
+  />
+</SdkCardGroup>
+
+## Client-side SDKs
+
+Knock's client-side helper libraries (also known as client-side SDKs) reduce the amount of work required to use Knock within your client-side applications and provide a foundation for [building in-app notification experiences](/in-app-ui/overview) on top of.
```
Comment (JEverhart383) [inline content/sdks/overview.mdx:52]:
```suggestion
Knock's client-side libraries (also known as client-side SDKs) reduce the amount of work required to use Knock within your client-side applications and provide a foundation for [building in-app notification experiences](/in-app-ui/overview) on top of.
```

**Review by JEverhart383 — APPROVED** [review]
Overall, this looks great. I like the visual treatment of the SDKs overview page with the cards. I also like how this separates out the API reference of the web SDKs from the actual component implementations which can now feel more like how-to guides, so I feel like this supports the divisions we want to make more clear in docs 2.0. 

I wonder if this SDK sidebar section shouldn't appear above developer tools. Looking at Google Analytics data, many of those React and JS reference pages get more visits than anything in the developer tools section. Something worth considering

Only downside of this re-org is adding another section the sidebar, really makes me want some horizontal nav structure to reduce user scrolling
Screenshot 2024-03-08 at 1 57 15 PM

**Pull request merged**